### PR TITLE
agnhost sync

### LINKED ISSUE #92059: agnhost e2e image inconsistent versions
I went to update the image usage after https://github.com/kubernetes/kubernetes/pull/87105 and discovered:
- `test/images/agnhost/VERSION` is `2.18`
- `test/images/agnhost/agnhost.go`'s root command has version `2.17`
- `test/utils/image/manifest.go` has agnhost versions `2.13` and `2.6`

/sig testing
/sig release
/kind bug
/assign

### PR DESCRIPTION
**Pull request by BenTheElder**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #92059 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by BenTheElder** [comment]
follow-up to #87105

**Comment by BenTheElder** [comment]
blocked on https://github.com/kubernetes/k8s.io/pull/958

**Comment by BenTheElder** [comment]
/assign johnbelamaric
This is a simple text substitution to catch up to redis's current, less offensive terminology.
The image change has been in, but the e2e tests have fallen behind the image.

**Comment by johnbelamaric** [comment]
Is `master` still the terminology for Redis? Should we use `primary`?

/approve

**Comment by BenTheElder** [comment]
master appears to still redis terminology, though this is only mildly redis-like...

I think we could change it but at the time I was focused on the "API" changes to the image that couldn't be done with a simple PR.

I'll double check what changing that would entail,
/hold

We do also still have some actual extremely dated redis stuff, but I don't think any of that is in conformance and I can approve most (all?) of it myself.

**Comment by BenTheElder** [comment]
... turns out that's going to require API changes too.

rolling up the first part of that here, since it also touches the conformance test descriptions
will file a follow-up to upgrade the image again and update the deployments again

File: test/conformance/testdata/conformance.yaml [hunk 1]
```diff
@@ -1397,11 +1397,11 @@
...
   description: Create a Pod running agnhost listening to port 6379. Using kubectl
-    expose the agnhost master replication controllers at port 1234. Validate that
+    expose the agnhost primary replication controllers at port 1234. Validate that
     the replication controller is listening on port 1234 and the target port is set
-    to 6379, port that agnhost master is listening. Using kubectl expose the agnhost
-    master as a service at port 2345. The service MUST be listening on port 2345 and
-    the target port is set to 6379, port that agnhost master is listening.
+    to 6379, port that agnhost primary is listening. Using kubectl expose the agnhost
+    primary as a service at port 2345. The service MUST be listening on port 2345 and
+    the target port is set to 6379, port that agnhost primary is listening.
```
Comment (BenTheElder) [inline test/conformance/testdata/conformance.yaml:1404]: This was done by hand, there are other references to "master" in the conformance tests that are talking about:
```
$ kubectl cluster-info
Kubernetes master is running at https://127.0.0.1:42939
KubeDNS is running at https://127.0.0.1:42939/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
```

Which is a very different change. 

_some_ of those are really referring to the `kubernetes.default` / apiserver service as the "master" service (not `kubectl cluster-info` output) and could be changed trivially, but I think that's a different PR.

File: test/e2e/kubectl/kubectl.go [hunk 2]
```diff
@@ -343,12 +343,12 @@ var _ = SIGDescribe("Kubectl client", func() {
 		forEachGBFile := func(run func(s string)) {
 			guestbookRoot := "test/e2e/testing-manifests/guestbook"
 			for _, gbAppFile := range []string{
-				"agnhost-slave-service.yaml",
+				"agnhost-replica-service.yaml",
 				"agnhost-master-service.yaml",
```
Comment (BenTheElder) [inline test/e2e/kubectl/kubectl.go:347]: I've already drafted the changes to these for after the e2e image is pushed

**Pull request merged**